rfcs/20211025-augru/augru-v2.md

```diff
@@ -0,0 +1,74 @@
+Proposal for GRU with Attentional Update V2 in DNNL
+===================================================
+
+## 1. Introduction
+
+[Original AUGRU RFC](README.md)
+
+OpenVino tnd TensorFlow teams reported that their AUGRU is different from the
+one defined in the original paper[[#1]][1]. The difference comes from attention
+update gate: 
+
+Original algorithm:
+![AUGRU-UPDATE-GATE](formula-augru-update.png)
+
+OpenVino and TensorFlow algorithm:
+![AUGRU-V2-UPDATE-GATE](formula-augru-v2-update.png)
+
+oneDNN implements the first version which prevents OpenVino and TensorFlow from
+utilizing oneDNN optimizations.
+
+
+## 2. Proposal
+
+### Option 1: Introduce new algorithm "augru_v2" which is aligned with OpenVino and TensorFlow
+
+Pros:
+- Support both original and modified definitions.
+- Algorithm fully defines cell behavior.
+
+Cons:
+- New RNN algorithm requires separate API (forward/backward, vanilla/LBR, C/C++):
+  - `dnnl_status_t dnnl_augru_v2_forward_desc_init()`
```

Comment:
This means the `v2` version will be supported along with the original one in the future, that is `v2` will not be turned into the regular one?

Reply:
yes, because it is not an extension, but a different algorithm. Indeed I had an idea to add augru_v2 and replace original augru in v3.0, which is kind of similar to the option 3 if we change augru behavior as a fixup in v2.6 (augru was introduced in v2.6). I asked frameworks to understand if there are any users of the original algorithm.

Reply:
Actually OpenVINO is going to define the specification for augru which includes both versions. We have no idea if someday somebody would use the one in the paper . So if you are going to delete the original one, please do it before openvino finalize its spec. If you could keep both, that would be better as openvino and onednn are aligned.

Reply:
Changed recommended solution to option 3.